source/MaterialXGenShader/ShaderTranslator.cpp

```diff
@@ -144,15 +144,18 @@ void ShaderTranslator::connectTranslationOutputs(NodePtr shader)
 
 void ShaderTranslator::translateShader(NodePtr shader, const string& destCategory)
 {
-    if (!shader)
+    _graph = nullptr;
+    _translationNode = nullptr;
+
+        if (!shader)
     {
         return;
     }
 
     const string& sourceCategory = shader->getCategory();
     if (sourceCategory == destCategory)
     {
-        throw Exception("The source shader \"" + shader->getNamePath() + "\" category is already \"" + destCategory + "\"");
+        return;
```

Comment:
Remove so as to not throw as it's not invalid. There is just nothing to do.
Removes false unit test failures if you run the tests twice as the translated file will throw here and fail the test.